src/main/java/org/broadinstitute/hellbender/utils/dataflow/DataflowUtils.java

```diff
@@ -84,6 +91,56 @@ public void processElement(ProcessContext c) {
     }
 
     /**
+     * Ingest a BAM file from a Hadoop file system and loads into a
+     * <code>PCollection<Read></code>.
+     * @param pipeline a configured Pipeline
+     * @param intervals intervals to select reads from
+     * @param bam Hadoop file path to read from
+     * @return a <code>PCollection<Read></code> with all the reads that overlap the
+     * given intervals in the BAM file
+     */
+    public static PCollection<Read> getReadsFromHadoopBam(final Pipeline pipeline, final List<SimpleInterval> intervals, final ValidationStringency stringency, final String bam) {
+        PCollection<KV<LongWritable, SAMRecordWritable>> input =
+            (PCollection<KV<LongWritable, SAMRecordWritable>>) pipeline.apply(
+                HadoopIO.Read.from(bam).withFormatClass(AnySAMInputFormat.class)
+                    .withKeyClass(LongWritable.class).withValueClass(SAMRecordWritable.class));
+        return input.apply(ParDo.of(new DoFn<KV<LongWritable, SAMRecordWritable>, Read>() {
+            @Override
+            public void processElement(ProcessContext c) throws Exception {
+                SAMRecord sam = c.element().getValue().get();
+                if (overlaps(sam, intervals)) {
+                    try {
+                        Read read = ReadConverter.makeRead(sam);
+                        c.output(read);
+                    } catch (SAMException e) {
+                        if (stringency == ValidationStringency.STRICT) {
+                            throw e;
+                        } else if (stringency == ValidationStringency.LENIENT) {
+                            logger.info("getReadsFromHadoopBam: " + e.getMessage());
+                        }
+                        // do nothing if silent
+                    }
+                }
+            }
+        }));
+    }
+
+    /**
+     * Tests if a given record overlaps any interval in a collection.
+     */
+    private static boolean overlaps(SAMRecord record, List<SimpleInterval> intervals) {
+        if (intervals == null || intervals.isEmpty()) {
+            return true;
```

Comment:
Shouldn't empty intervals not overlap?  

In most places in Hellbender we treat an empty interval list as unspecified and therefore including the whole genome.